server/opts.go

```diff
@@ -894,6 +925,155 @@ func parseGateway(v interface{}, o *Options, errors *[]error, warnings *[]error)
 	return nil
 }
 
+// parseCluster will parse the cluster config.
+func parseLeafNodes(v interface{}, opts *Options, errors *[]error, warnings *[]error) error {
+	tk, v := unwrapValue(v)
+	cm, ok := v.(map[string]interface{})
+	if !ok {
+		return &configErr{tk, fmt.Sprintf("Expected map to define a leafnode, got %T", v)}
+	}
+
+	for mk, mv := range cm {
+		// Again, unwrap token value if line check is required.
+		tk, mv = unwrapValue(mv)
+		switch strings.ToLower(mk) {
+		case "listen":
+			hp, err := parseListen(mv)
+			if err != nil {
+				err := &configErr{tk, err.Error()}
+				*errors = append(*errors, err)
+				continue
+			}
+			opts.LeafNode.Host = hp.host
+			opts.LeafNode.Port = hp.port
+		case "port":
+			opts.LeafNode.Port = int(mv.(int64))
+		case "host", "net":
+			opts.LeafNode.Host = mv.(string)
+		case "authorization":
+			auth, err := parseAuthorization(tk, opts, errors, warnings)
+			if err != nil {
+				*errors = append(*errors, err)
+				continue
+			}
+			if auth.users != nil {
+				err := &configErr{tk, fmt.Sprintf("Leafnode authorization does not allow multiple users")}
+				*errors = append(*errors, err)
+				continue
+			}
+			opts.LeafNode.Username = auth.user
+			opts.LeafNode.Password = auth.pass
+			opts.LeafNode.AuthTimeout = auth.timeout
+
+			if auth.defaultPermissions != nil {
```

Comment:
Probably taken from parseCluster but don't think this applies here..

Reply:
I think we will eventually want this for DES, but I can remove for now.

Reply:
Remove for now, especially that if you look at this code it says that it is deprecated. This was when I had originally put the route perms inside auth. But I have then moved them out of auth {}.

Reply:
ok will do.